File: TP2PARTIE2/app/src/main/java/umontpellier/gl1/tp2partie2/MainActivity.java
package umontpellier.gl1.tp2partie2;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private PaysAdapter paysAdapter;
    private DatabaseHelperPays dbHelper;

    private DatabaseHelper dbHelperUser;
    private ImageView profileImageView;

    private String currentUserEmail; // Email de l'utilisateur connecté

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dbHelper = new DatabaseHelperPays(this);
        dbHelperUser = new DatabaseHelper(this);


        currentUserEmail = getSharedPreferences("user_prefs", MODE_PRIVATE).getString("user_email", null);



        recyclerView = findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));


        List<Pays> paysList = dbHelper.getAllPays();


        paysAdapter = new PaysAdapter(this, paysList,dbHelper,false);
        recyclerView.setAdapter(paysAdapter);


        CardView cardView = findViewById(R.id.cardCart);
        cardView.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, HomeActivity.class);
            startActivity(intent);
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        // Rafraîchir la liste des pays à chaque retour sur l'écran
        List<Pays> paysList = dbHelper.getAllPays();
        paysAdapter = new PaysAdapter(this, paysList,dbHelper,false);
        recyclerView.setAdapter(paysAdapter);
    }




}

File: TP2App1/app/src/main/java/umontpellier/gl1/tp2app1/MainActivity.kt
package umontpellier.gl1.tp2app1

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import kotlin.system.exitProcess

class MainActivity : AppCompatActivity() {

    private var currentUserEmail: String? = null // Email de l'utilisateur connecté
    private lateinit var txtName: TextView
    private lateinit var txtEmail: TextView
    private lateinit var dbHelperUser: DatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Initialisation de DatabaseHelper
        dbHelperUser = DatabaseHelper(this)

        // Récupération de l'email de l'utilisateur connecté depuis SharedPreferences
        val sharedPreferences: SharedPreferences = getSharedPreferences("user_prefs", MODE_PRIVATE)
        currentUserEmail = sharedPreferences.getString("user_email", null)

        // Initialisation des vues
        txtName = findViewById(R.id.txtname)
        txtEmail = findViewById(R.id.emailtxt)

        // Charger les informations de l'utilisateur
        loadUserInfo()

        // Associer chaque CardView à son ID et définir un écouteur de clic
        findViewById<CardView>(R.id.cardCart).setOnClickListener {
            startActivity(Intent(this, Exercice1::class.java))
        }

        findViewById<CardView>(R.id.cardMap).setOnClickListener {
            startActivity(Intent(this, Ex2Activity::class.java))
        }

        findViewById<CardView>(R.id.cardInbox).setOnClickListener {
            startActivity(Intent(this, Ex3Activity::class.java))
        }

        findViewById<CardView>(R.id.cardSetting).setOnClickListener {
            startActivity(Intent(this, Ex4Activity::class.java))
        }

        findViewById<CardView>(R.id.cardFavourite).setOnClickListener {
            startActivity(Intent(this, Ex5Activity::class.java))
        }

        findViewById<CardView>(R.id.cardLogout).setOnClickListener {
            startActivity(Intent(this, Ex6Activity::class.java))
        }

        findViewById<CardView>(R.id.cardNew1).setOnClickListener {
            startActivity(Intent(this, Ex7Activity::class.java))
        }

        findViewById<CardView>(R.id.cardNew2).setOnClickListener {
            startActivity(Intent(this, LoginScreen::class.java))
        }


    }

    private fun loadUserInfo() {
        if (currentUserEmail != null) {
            val userName = dbHelperUser.getUserName(currentUserEmail!!)
            if (userName != null) {
                txtName.text = userName
            } else {
                txtName.text = "Nom non trouvé"
            }
            txtEmail.text = currentUserEmail
        } else {
            Log.d("MainActivity", "No user email found")
        }
    }
}